### PR TITLE
Item details component

### PR DESCRIPTION
**Pull request by ashatat**
Details on #66 

Props is object
ex: 
```jsx
// call the component
<ItemDetails {...fakeData} />

// fake data object
fakeData = {
  id: '1',
  userId: '2',
  name: 'Plain Backpack',
  description:
    'Lorem ipsum dolor sit amet isse potenti. Vesquam ante aliquet lacusemper elit. Cras neque nulla, convallis non commodo et, euismod nonsese. At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum.',
  category: 'Accessories',
  sku: '67890',
  status: 'sale',
  details: [
    {
      id: '1',
      pricing: {
        price: '305',
        discount_price: '0',
      },
      color: 'Blue',
      quantity: 10,
      tags: [
        {
          id: '1',
          name: 'Backpack',
        },
        {
          id: '2',
          name: 'Accessories',
        },
      ],
    },
    {
      id: '2',
      pricing: {
        price: '360',
        discount_price: '50',
      },
      color: 'Green',
      quantity: 6,
      tags: [
        {
          id: '1',
          name: 'Backpack',
        },
        {
          id: '2',
          name: 'Accessories',
        },
      ],
    },
  ],
};

```

### CODE REVIEW

**Comment by amusameh** [comment]
You need to add all your components in one folder

File: client/src/components/ItemDetails/index.jsx [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+        id: '2',
+        name: 'Accessories',
+      },
+    ],
+  };
+
+  addToCartButton = {
+    adding: (
+      <React.Fragment>
+        <FaCircleNotch className="item-details__add-to-cart--icon-spin" />{' '}
```
[the remark this answers is not in the document]
Comment (ashatat) [reply client/src/components/ItemDetails/index.jsx]: Just to add a space between the spinner and the ADDING word @amusameh

File: client/src/components/ItemDetails/index.jsx [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    cart: 'default',
+    wList: 'default',
+    color: '',
+    quantity: 0,
+    tags: [],
+  };
+
+  addToCartButton = {
+    adding: (
+      <React.Fragment>
```
Comment (FarahZaqout) [inline client/src/components/ItemDetails/index.jsx]: Should destructure this.
```suggestion
      <Fragment>
```

File: client/src/components/ItemDetails/index.jsx [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    adding: <FaCircleNotch className="item-details__add-to-cart--icon-spin" />,
+    added: <MdDone />,
+    default: (
+      <IconContext.Provider value={{ size: '25px' }}>
+        <MdPlaylistAdd />
+      </IconContext.Provider>
+    ),
+  };
+
+  prevItem = () => {
```
Comment (FarahZaqout) [inline client/src/components/ItemDetails/index.jsx]: ¿Que?

I don't understand how this works.

File: client/src/components/ItemDetails/index.jsx [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+                  className="item-details__qty-field"
+                  onChange={this.handelChange}
+                />
+                <button
+                  onClick={this.handelIncrement}
+                  type="button"
+                  className="item-details__qty-inc"
+                >
+                  <MdKeyboardArrowUp />
+                </button>
```
[the remark this answers is not in the document]
Comment (ashatat) [reply client/src/components/ItemDetails/index.jsx]: Thanks, I didn't notice this problem till I open it with google chrome.

File: client/src/components/ItemDetails/index.jsx (hunk 2 above)
Comment (ashatat) [reply client/src/components/ItemDetails/index.jsx]: I did it :+1:

File: client/src/components/ItemDetails/index.jsx (hunk 3 above)
Comment (ashatat) [reply client/src/components/ItemDetails/index.jsx]: how what works? @FarahZaqout

File: client/src/components/ItemDetails/ItemDetails.css [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+  font-size: 14px;
+  font-weight: bold;
+  letter-spacing: 1px;
+  color: #999;
+  background: none;
+  border: 2px solid #999;
+  transition: all ease-in-out 1s;
+}
+
+.item-details__add-to-cart:hover, .item-details__add-to-cart:focus {
```
Comment (RamyAlshurafa) [inline client/src/components/ItemDetails/ItemDetails.css]: > Write one selector per line

for [more](https://10up.github.io/Engineering-Best-Practices/css/)

File: client/src/components/ItemDetails/index.jsx [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+      this.setState(prevState => ({ quantity: prevState.quantity - 1 }));
+    }
+  };
+
+  addToCart = () => {
+    const { details, userId, status } = this.props;
+    const { color, quantity } = this.state;
+    if (color && quantity && (!status || status === 'sale')) {
+      this.setState({ cart: 'adding' });
+      fetch('/api/v1/add-to-cart', {
```
Comment (RamyAlshurafa) [inline client/src/components/ItemDetails/index.jsx]: I prefer to use `'axios'`
```suggestion
     axios.post('/api/v1/add-to-cart', {
           userId,
          itemDetailId: details[Number(color)].id,
          quantity,
  })
  .then(...
```
, it's much easier, so no need for `credentials`, `JSON.stringify`, `headers`

Comment (ashatat) [reply client/src/components/ItemDetails/index.jsx]: but I think we all used `fetch`

**Pull request merged**